src/hevm/src/EVM.hs

```diff
@@ -1880,37 +1883,82 @@ type CheatAction = Word -> Word -> Buffer -> EVM ()
 cheatActions :: Map Word32 CheatAction
 cheatActions =
   Map.fromList
-    [ action "warp(uint256)" $
-        \sig _ _ input -> case decodeStaticArgs input of
+    [ action "warp(uint256)" [AbiUIntType 256] $
+        \sig _ _ _ input -> case decodeStaticArgs input of
           [x]  -> assign (block . timestamp) (mksym x)
           _ -> vmError (BadCheatCode sig),
 
-      action "roll(uint256)" $
-        \sig _ _ input -> case decodeStaticArgs input of
+      action "roll(uint256)" [AbiUIntType 256] $
+        \sig _ _ _ input -> case decodeStaticArgs input of
           [x] -> forceConcrete (mksym x) (assign (block . number))
           _ -> vmError (BadCheatCode sig),
 
-      action "store(address,bytes32,bytes32)" $
-        \sig _ _ input -> case decodeStaticArgs input of
+      action "store(address,bytes32,bytes32)" [AbiAddressType, AbiBytesType 32, AbiBytesType 32] $
+        \sig _ _ _ input -> case decodeStaticArgs input of
           [a, slot, new] ->
             makeUnique (mksym $ sFromIntegral a) $ \(C _ (num -> a')) ->
               fetchAccount a' $ \_ -> do
                 modifying (env . contracts . ix a' . storage) (writeStorage (mksym slot) (mksym new))
           _ -> vmError (BadCheatCode sig),
 
-      action "load(address,bytes32)" $
-        \sig outOffset _ input -> case decodeStaticArgs input of
+      action "load(address,bytes32)" [AbiAddressType, AbiBytesType 32] $
+        \sig _ outOffset _ input -> case decodeStaticArgs input of
           [a, slot] ->
             makeUnique (mksym $ sFromIntegral a) $ \(C _ (num -> a'))->
               accessStorage a' (mksym slot) $ \res -> do
                 assign (state . returndata . word256At 0) res
                 assign (state . memory . word256At outOffset) res
+          _ -> vmError (BadCheatCode sig),
+
+      action "sign(uint256,bytes)" [AbiUIntType 256, AbiBytesDynamicType] $
+        \sig tps outOffset _ input -> case decodeBuffer tps input of
+          CAbi [AbiUInt 256 sk, AbiBytesDynamic msg] -> let
+              curve = getCurveByName SEC_p256k1
+              priv = PrivateKey curve (num sk)
+              s = ethsign priv msg
+              v = if (sign_s s) % 2 == 0 then 27 else 28
+              encoded = encodeAbiValue $
+                AbiTuple (RegularVector.fromList
+                  [ AbiUInt 8 v
+                  , AbiBytes 32 (encodeAbiValue . AbiUInt 256 . fromInteger $ sign_r s)
+                  , AbiBytes 32 (encodeAbiValue . AbiUInt 256 . fromInteger $ sign_s s)
+                  ])
+            in do
+              assign (state . returndata) (ConcreteBuffer encoded)
+              copyBytesToMemory (ConcreteBuffer encoded) (num . BS.length $ encoded) 0 outOffset
+          _ -> vmError (BadCheatCode sig),
```

Comment:
One thing that's a little weird here is that we throw a `BadCheatCode` instead of an  `UnexpectedSymbolicArg` in case someone tries to call `hevm.sign` with a symbolic argument...

Reply:
no longer valid....